discoursemap/modules/vulnerability_module.py

```
@@ -275,27 +331,114 @@
+        start_time = time.time()
+        response = self.scanner.make_request(url, params={'q': payload})
+        response_time = time.time() - start_time
+        
+        # If response time is significantly longer than baseline, likely time-based SQLi
+        time_threshold = 2.5  # seconds
+        if response_time > baseline_time + time_threshold:
+            return True
+            
+        return False
+    
     def _extract_sql_evidence(self, response_text, payload):
```

Ruff (0.12.2)

Unused method argument: `payload`

(ARG002)